src/test_runner/failure_persistence/file.rs

```diff
@@ -144,159 +233,118 @@ where
+                    (corrupt file or newer proptest version?)",
+                    &path.display(), lineno + 1, parts[0]);
+            return None;
+        },
+    };
+
+    // Parse using the chosen one:
+    if let Ok(seed) = parser(&parts[1..]) {
+        Some(seed)
+    } else {
+        eprintln!("proptest: {}:{}: unparsable line, ignoring",
+                    path.display(), lineno + 1);
+        None
+    }
+}
 
-                let input = io::BufReader::new(fs::File::open(path)?);
-                for (lineno, line) in input.lines().enumerate() {
-                    let mut line = line?;
-                    if let Some(comment_start) = line.find('#') {
-                        line.truncate(comment_start);
-                    }
+fn parse_seed(parts: &[&str]) -> Result<[u8; 16], ParseIntError> {
+    let mut ret = [0; 16];
+    for (src, dst) in parts.iter().zip(ret.iter_mut()) {
+        *dst = src.parse()?;
+    }
+    Ok(ret)
+}
 
-                    let parts = line.trim().split(' ').collect::<Vec<_>>();
-                    if 5 == parts.len() && "xs" == parts[0] {
-                        if let Ok(seed) = parse_seed(&*parts) {
-                            ret.push(seed);
-                        } else {
-                            eprintln!(
-                                "proptest: {}:{}: unparsable line, \
-                             ignoring",
-                                path.display(),
-                                lineno + 1
-                            );
-                        }
-                    } else if parts.len() > 1 {
-                        eprintln!(
-                            "proptest: {}:{}: unknown case type `{}` \
-                         (corrupt file or newer proptest version?)",
-                            &path.display(),
-                            lineno + 1,
-                            parts[0]
-                        );
-                    }
-                }
+fn parse_seed_old(parts: &[&str]) -> Result<[u8; 16], ParseIntError> {
+    let mut ret = [0u32; 4];
+    for (src, dst) in parts.iter().zip(ret.iter_mut()) {
+        *dst = src.parse()?;
+    }
 
-                Ok(ret)
-            },
-        );
+    Ok(convert_to_new_format(ret))
+}
 
-        match result {
-            Ok(r) => r,
-            Err(err) => {
-                if io::ErrorKind::NotFound != err.kind() {
-                    eprintln!(
-                        "proptest: failed to open {}: {}",
-                        &path.map(|x| &**x)
-                            .unwrap_or_else(|| Path::new("??"))
-                            .display(),
-                        err
-                    );
-                }
-                vec![]
-            }
+fn convert_to_new_format(old_format: [u32; 4]) -> [u8; 16] {
+    use byteorder::{NativeEndian, ByteOrder};
+    let mut new_format = [0; 16];
+    NativeEndian::write_u32_into(&old_format[..], &mut new_format);
```

Comment:
Does `rand` actually convert the seed into `[u32;4]` using native byte order? If so, it'd probably be better to keep storing the `u32`s since otherwise they're not portable.

Reply:
Don't think rand does that; It seems to convert back into `[u32; 4]` using little endian; https://docs.rs/rand/0.5.0/src/rand/prng/xorshift.rs.html#89-130 (`rand` uses little endian everywhere now apparently...)

`le::read_u32_into` is defined here: https://github.com/rust-lang-nursery/rand/blob/591519b5a6d5b8ae14f7bb536cb3439449ad8092/rand_core/src/le.rs#L18-L39